environs/bootstrap/bootstrap.go

```diff
@@ -655,19 +655,52 @@ func isCompatibleVersion(v1, v2 version.Number) bool {
 	return v1.Compare(v2) == 0
 }
 
-// setPrivateMetadataSources sets the default tools metadata source
-// for tools syncing, and adds an image metadata source after verifying
-// the contents.
+// setPrivateMetadataSources verifys the metadataDir and uses it to
+// set the default tools metadata source for tools syncing, and adds
+// an image metadata source after verifying the contents.  If the
+// directory ends in tools, only the default tools metadata source
+// will be set.  Same for images.
 func setPrivateMetadataSources(metadataDir string) ([]*imagemetadata.ImageMetadata, error) {
-	logger.Infof("Setting default tools and image metadata sources: %s", metadataDir)
-	tools.DefaultBaseURL = metadataDir
+	if _, err := os.Stat(metadataDir); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Annotate(err, "cannot access simplestreams metadata directory")
+		}
+		return nil, errors.NotFoundf("simplestreams metadata source: %s", metadataDir)
+	}
 
-	imageMetadataDir := filepath.Join(metadataDir, storage.BaseImagesPath)
+	toolsMetadataDir := metadataDir
+	ending := filepath.Base(toolsMetadataDir)
+	if ending != storage.BaseToolsPath {
+		toolsMetadataDir = filepath.Join(metadataDir, storage.BaseToolsPath)
+	}
+	if _, err := os.Stat(toolsMetadataDir); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Annotate(err, "cannot access tools metadata")
+		}
+		logger.Debugf("default tools metadata source: %s", tools.DefaultBaseURL)
+	} else {
+		if ending == storage.BaseToolsPath {
+			tools.DefaultBaseURL = filepath.Dir(metadataDir)
+			logger.Debugf("setting default tools metadata source: %s", tools.DefaultBaseURL)
```

Comment:
agent binaries

especially given the config option would be "agent-metadata-url"